app/index.js

```diff
@@ -1,13 +1,11 @@
-const server = require('./server')
+const startServer = require('./server/start-server')
+const unhandledRejection = require('./exception/unhandled-rejection')
+const { start: startMessageService, stop: stopMessageService } = require('./messaging/service')
 
-const init = async () => {
-  await server.start()
-  console.log('Server running on %s', server.info.uri)
-}
+startServer()
+startMessageService()
 
-process.on('unhandledRejection', (err) => {
-  console.error(err)
-  process.exit(1)
+process.on('unhandledRejection', async (err) => {
+  await stopMessageService()
```

Comment:
We need to check and confirm that any unhandled exception should stop messaging service.

Reply:
Done - this was a tricky test 😆